### ISSUE TITLE
[P2] Disable autograd in MERT evaluation and bound embedding memory across full songs

### ISSUE THREAD

**Issue by KadoBOT** [KadoBOT/ComfyUI-YuE2-Voices#16]
## Scope
Whole-codebase audit of `29c0e939261067afae1ad517f23aab50d62648db`. New concrete evaluator defect, complementary to the metric-validity work in #4 and regression infrastructure in #15.

Source tracing and an isolated CPU reproduction with the exact `embed` loop and a small trainable stand-in model established that evaluation outputs retain autograd graphs. No real MERT/CUDA out-of-memory benchmark was run; actual peak memory and failure thresholds remain to be measured.

## Evidence
[`checks/mert_identity.py::embed / main`](https://github.com/KadoBOT/ComfyUI-YuE2-Voices/blob/29c0e939261067afae1ad517f23aab50d62648db/checks/mert_identity.py) loads MERT with `.eval()`, but never disables gradients or freezes its parameters. `embed()` wraps inference only in `torch.autocast`, which is a precision context, not a gradient guard.

Each chunk's `hidden.reshape(-1, 1024).float().mean(0)` is appended to `pooled`; all chunk graphs remain reachable until the stacked embedding is released. `reference_embedding` then remains live throughout the comparison loop. The previous candidate embedding is also live while the next right-hand-side computation executes. Chunking the waveform therefore does not bound retained activation memory to one chunk.

This differs from `worker/voice_trainer.py::extract_tokens`, which correctly uses `@torch.no_grad()`, and `checks/prep_artist.py::mert_l20`, which explicitly uses inference mode.

## Reproduction result
Using a 61-second synthetic waveform, the current `embed` loop and a CPU `nn.Module` with a learnable parameter:

```text
model.training                         False
current embedding.requires_grad       True
current embedding.grad_fn             MeanBackward1
embedding under torch.inference_mode  requires_grad=False
```

The CUDA autocast context was disabled by PyTorch on the CPU-only test; that does not alter the gradient-recording issue. This is a contract reproduction, not a claim to have run the real music encoder.

## Why it matters
The project needs repeatable, multi-song/multi-seed validation before approving a reusable vocalist. Retaining training graphs in a read-only evaluator wastes VRAM and can make larger reference sets fail for reasons unrelated to identity quality. A 30-second chunk loop should not accumulate every chunk's full forward graph.

## Requested fix
Use `torch.inference_mode()` for the entire embedding/reference evaluation path, and freeze MERT parameters as an additional explicit invariant. Keep the numeric preprocessing/score changes separate from this memory-only fix so score equivalence can be checked. Consider incremental, explicitly weighted pooling rather than retaining per-chunk outputs; preserve a documented weighting rule.

## Acceptance criteria
- [ ] Reference and candidate embeddings have `requires_grad=False` and `grad_fn is None` in the actual CLI path.
- [ ] A stand-in-model unit test fails against the current code and passes with the gradient guard.
- [ ] No parameter gradients are populated, and no hidden-state graph remains reachable between candidate evaluations.
- [ ] A GPU memory test compares multiple clip lengths and repeated candidates after warm-up; memory growth is explained by necessary inputs/results, not accumulated forward graphs.
- [ ] On identical input/model/dtype settings, scores before and after the memory-only change agree within a declared tolerance.
- [ ] Handle sub-second/empty accepted inputs explicitly, alongside the boundary work in #12, rather than failing inside `torch.stack`.
